app/controllers/v0/sessions_controller.rb

```diff
@@ -116,8 +116,20 @@ def saml_user
       @saml_user ||= SAML::User.new(@saml_response)
     end
 
+    # This is used in persist_session_and_user
     def new_user_from_saml
-      @new_user_from_saml ||= User.new(saml_user.to_hash)
+      @new_user_from_saml ||= begin
+        if saml_user.mhv?
+          if saml_user.icn.present?
+            # fetch name and other attributes from MVI and use in app/models/user.rb
+          else
+            # either raise an error here and log # of times this happens or attempt to load user as loa3
```

Comment:
Questions are more me understanding this than comments on the approach:
1) Is another option to log that this happened so we can track in sentry, but not raise an exception for now? Then revisit if we see this happening in production
2) What are the possible results of "attempting" to load user as LOA3:
  - is there some signal that would indicate this wasn't successful?
  - if so, then would we try to load user as LOA3 and if that fails, raise an exception?

Reply:
Currently if user is not "LOA3" we don't fetch from MVI at all, so I guess I was worried about deviating too far away from that current paradigm. It definitely should not be done if the user is not "premium" or the "new LOA3." But I suppose if they are LOA3 and they don't have an ICN whereby we could query for basic fields such as name, dob, gender, ssn, I was worried what the ramifications might be seeing as some of these fields are validated on user.rb as needing to be present.

To answer your questions:
1. I think we know with certainty that this will happen for some 13% of MHV users.
2. I don't know what the ramifications are. I imagine the F/E might have built up some expectation that LOA3 implies these basic values being present. If they don't have proper null handling, things could break in unexpected ways. Specs across vets-api would also likely need to be re-worked.

I'm not sure what you mean by "signal." If the saml response has an authn_context that signifies it is an MHV sign-in user and MHV characterizes it as "premium" we would need to have an ICN to query MVI or else the experience for this LOA3 could be at minimum severely limited, or at most error prone.

I'm certainly open to trying it out and seeing how it goes on staging, but wouldn't feel comfortable pushing that approach on to production. That being said, I dont know if I'm personally on a position to say that some 13% of total MHV users that *could* be affected is reason enough to delay launching?

Reply:
I think Alastair has set aside some time on the calendar tomorrow for us to discuss the evolution of identity and how tightly current_user is coupled with MVI, etc. We can probably wait and discuss this and potential ramifications tomorrow.

Reply:
sorry to have missed Alastair's session! I saw that on the agenda, but have been away all week.  In any case, @patrickvinograd should review since I don't want to block it and definitely want to spend some time understanding the expectations shared across vets-api and vets-website. I'll catch up with Alastair to hear what he shared